table.cc

```diff
@@ -1958,6 +1958,7 @@ future<int64_t>
 table::disable_sstable_write() {
     _sstable_writes_disabled_at = std::chrono::steady_clock::now();
     return _sstables_lock.write_lock().then([this] {
+      return _sstable_deletion_sem.wait().then([this] {
         if (_sstables->all()->empty()) {
```

Comment:
Please document the lock order in a follow up.

Perhaps we should fold _sstable_deletion_sem into _sstables_lock.

Reply:
documented in 0e4567c8815e4bc7f6acc72aee88194e75a2107f

Regarding folding into `_sstables_lock` , they are currently taken at different times in the table::compact_sstables -> on_compaction_completion path, and I'm not sure about the table::snapshot path.  I think that keeping them separate makes sense.